beetsplug/absubmit.py

```
@@ -0,0 +1,159 @@
+
+    def _get_analysis(self, item):
+        mbid = item['mb_trackid']
+        # If file has no mbid skip it.
+        if not mbid:
+            self._log.info('Not analysing {}, missing '
+                           'musicbrainz track id.', item)
+            return None
+        # If file format is not supported skip it.
+        if item['format'].lower() not in self.supported_formats:
+            self._log.info('Not analysing {}, file not in '
+                           'supported format.', item)
+            return None
+
+        # Temporary file to save extractor output to.
+        tmp_file, filename = tempfile.mkstemp(suffix='.json')
+        try:
+            # Close the file, so the extractor can overwrite it.
+            call([self.extractor, util.syspath(item.path), filename])
+            with open(filename) as tmp_file:
+                analysis = json.loads(tmp_file.read())
+            # Calculate extractor hash.
+            m = hashlib.sha1()
+            with open(self.extractor, 'rb') as extractor:
+                m.update(extractor.read())
+            # Add the hash to the output.
+            analysis['metadata']['version']['essentia_build_sha'] = \
+                m.hexdigest()
+            return analysis
+        finally:
+            try:
+                os.remove(filename)
+            except OSError as e:
+                # errno 2 means file does not exist, just ignore this error.
+                if e.errno != 2:
+                    raise
+
+    def _submit_data(self, item, data):
+        mbid = item['mb_trackid']
+        headers = {'Content-Type': 'application/json'}
+        response = requests.post(self.base_url.format(mbid=mbid),
+                                 json=data, headers=headers)
+        # Test that request was successful and raise an error on failure.
+        if response.status_code != 200:
+            try:
+                message = response.json()['message']
+            except Exception as e:
```

I'm also not very sure if we should raise an error here, or just skip uploading the file.